src/main/scala/edu/ie3/powerFactory2psdm/model/powerfactory/GridModel.scala

```diff
@@ -87,13 +111,56 @@ object GridModel extends LazyLogging {
     val lines = rawLines.map(line => Line.build(line))
     val lineTypes = rawLineTypes.map(LineType.build)
     val switches = rawSwitches.flatMap(Switch.maybeBuild)
+    val trafoTypes2W = rawTrafoTpyes2W.map(TransformerType2W.build)
 
     GridModel(
       nodes,
       lineTypes,
       lines,
-      switches
+      switches,
+      trafoTypes2W,
+      conversionPrefixes
     )
   }
 
+  private def extractProjectSettings(
+      rawSettings: Option[List[ProjectSettings]]
+  ): ProjectSettings = {
+    rawSettings match {
+      case Some(List(settings)) => settings
+      case Some(List(_, _, _*)) =>
+        throw ConversionException(
+          "There are multiple project settings defined."
+        )
+      case None =>
+        throw ConversionException("There are no project settings defined.")
+    }
+  }
+
+  private def checkUnitSystem(unitSystem: Option[Double]): Unit = {
+    unitSystem match {
+      case Some(0) => ()
```

Comment:
Would suggest to introduce an enum, that is given some more insight into what "0" actually means and what other values are apparent.